girder/utility/s3_assetstore_adapter.py

```diff
@@ -406,14 +409,10 @@ def stream():
                         yield chunk
             return stream
 
-    def importData(self, parent, parentType, params, progress, user, **kwargs):
+    def importData(self, parent, parentType, params, progress, user, recursive=True, **kwargs):
         importPath = params.get('importPath', '').strip().lstrip('/')
-
-        if importPath and not importPath.endswith('/'):
-            importPath += '/'
-
         bucket = self.assetstore['bucket']
-        paginator = self.client.get_paginator('list_objects')
+        paginator = self.client.get_paginator('list_objects_v2')
```

Comment:
This will not work if using Google Cloud Storage. Is it strictly necessary to switch to v2 for what you need?

Reply:
I've tested it using google cloud storage and it appears to work.  Is there something saying it doesn't?

Reply:
Weird, there appears to be more nuance here.  These lines work, but https://issuetracker.google.com/issues/155109631 does not work.  I don't know what the difference is, but I'll just revert to list_objects.

Reply:
If it's as simple as just using `list_objects` instead of `list_objects_v2`, let's do that. I'm not sure of all the gory details, I just know that last week when I wrote a very basic script to list objects in a GCS bucket using this same paginator API, it didn't work with `v2` but worked fine with `list_objects`. There are some slight configuration differences that could account for it. 🤷